firmware/fw_base.S

```diff
@@ -124,6 +126,10 @@ _wait_for_boot_hart:
 	REG_L	a5, (a4)
 	beqz	a5, _wait_for_boot_hart
 
+	csrr	a6, CSR_MHARTID
+	beqz	a6, _start_warm
+	li	ra, 0
+	call	_reset_regs
```

Comment:
I think you should unconditionally reset all registers in _start_warm() for all HARTs.

Reply:
ok. Updated and created another PR #80